mllib/src/test/scala/org/apache/spark/ml/tree/impl/TreeTests.scala

```diff
@@ -124,8 +129,8 @@ private[ml] object TreeTests extends SparkFunSuite {
    *       make mistakes such as creating loops of Nodes.
    */
   private def checkEqual(a: Node, b: Node): Unit = {
-    assert(a.prediction === b.prediction)
-    assert(a.impurity === b.impurity)
+    assert(a.prediction ~== b.prediction absTol 1e-8)
+    assert(a.impurity ~== b.impurity absTol 1e-8)
```

Comment:
All over the test suites we use tolerances as literal doubles instead of making a variable for each and every one. I think it would be over-engineering to do this.